runtime/src/chpl-topo.c

```diff
@@ -37,6 +38,9 @@
 #include <string.h>
 #include <unistd.h>
 
+#if defined(CHPL_HWLOC_HWLOC) || defined(CHPL_HWLOC_SYSTEM)
+#define CHPL_HAS_HWLOC
```

Comment:
Why bother defining `CHPL_HAS_HWLOC`at all now? Just replace it's use with `defined(CHPL_HWLOC_HWLOC) || defined(CHPL_HWLOC_SYSTEM)`? (and update the closing `#endif` comment)